### PR TITLE
Feat: Allow mixing configs & account resources

### PR DESCRIPTION
**Pull request by Laubi**
#### What this PR does / Why we need it:
This PR introduces the possibility of mixing configs and account resources in the same project.

Warnings are printed if configs of the other types are encountered, as we discourage that behavior.
Whenever a config has both configs and AM resources defined, we print an error as this is forbidden.

### Deploying AM resources with config resources also defined in the project
```
PS > monaco account deploy .\manifest.yaml
2023-11-28T16:47:34+01:00       info    Loading manifest ".\\manifest.yaml". Restrictions: groups=[], environments=[]
2023-11-28T16:47:34+01:00       warn    File "accounts-configs\\config.yaml" appears to be an config file, skipping loading                                                    # THIS LINE
2023-11-28T16:47:37+01:00       info    Deploying policy My Other Policy to account my-account...
2023-11-28T16:47:38+01:00       info    Deploying group My Other Group to account my-account...
2023-11-28T16:47:38+01:00       info    Updating policy bindings and permissions...
2023-11-28T16:47:39+01:00       info    Deploying user david@invalid.dynatrace.com to account my-account...
2023-11-28T16:47:40+01:00       info    Updating group bindings...
```

### Deploying configs with AM resources also defined
```
PS > monaco deploy .\manifest.yaml
2023-11-28T16:47:48+01:00       info    Loading 1 projects...
2023-11-28T16:47:48+01:00       warn    File "accounts-configs\\accounts.yaml" appears to be an account-config file, skipping loading                                      # THIS LINE
2023-11-28T16:47:48+01:00       info    Projects to be deployed (1):
2023-11-28T16:47:48+01:00       info      - accounts-configs
2023-11-28T16:47:48+01:00       info    Environments to deploy to (2):
2023-11-28T16:47:48+01:00       info      - classic
2023-11-28T16:47:48+01:00       info      - platform
…
```
### Deploying AM resources with configs in the same file
```
PS > monaco account deploy .\manifest.yaml
…
2023-11-28T16:59:18+01:00       error   Error: failed to load all account management resources: failed to load account managment resources from accounts-configs: failed to parse file "accounts-configs\\accounts.yaml": mixing both configurations and account resources is not allowed
```

### Deploying configs with account resources in the same file
```
…
2023-11-29T13:52:01+01:00       info    Loading 1 projects...
2023-11-29T13:52:01+01:00       error   Failed to load projects - 1 errors occurred:
2023-11-29T13:52:01+01:00       error   failed to load config from file "accounts-configs\\config.yaml": mixing both configurations and account resources is not allowed
2023-11-29T13:52:01+01:00       error   Error: failed to load projects - 1 errors occurred
2023-11-29T13:52:01+01:00       error   error logs written to .logs\20231129-135200-errors.log
```

### CODE REVIEW

File: pkg/persistence/account/loader/load_test.go [hunk 1]
```diff
@@ -145,11 +153,21 @@ func TestValidateReferences(t *testing.T) {
...
+			path:     "testdata/configs-accounts-mixed.yaml",
+			expected: ErrMixingConfigs,
+		},
 		{
 			name:     "valid",
 			path:     "testdata/valid.yaml",
 			expected: nil,
 		},
+		{
+			name:     "valid loading configs",
```
Comment (UnseenWizzard) [inline pkg/persistence/account/loader/load_test.go:167]: test this explitily please like the test on line 70 - then it's clear from looking at the test, that not only do we want no error, but also 0 groups, policies and users

File: pkg/persistence/config/loader/config_loader.go [hunk 2]
```diff
@@ -86,25 +114,21 @@ func LoadConfig(fs afero.Fs, context *LoaderContext, filePath string) ([]config.
...
-		return nil, err
-	}
+func parseFile(data []byte) ([]persistence.TopLevelConfigDefinition, error) {
 
 	definition := persistence.TopLevelDefinition{}
-	err = yaml.UnmarshalStrict(data, &definition)
+	err := yaml.UnmarshalStrict(data, &definition)
 
 	if err != nil {
 		if strings.Contains(err.Error(), fmt.Sprintf("field config not found in type %s", persistence.GetTopLevelDefinitionYamlTypeName())) {
```
Comment (UnseenWizzard) [inline pkg/persistence/config/loader/config_loader.go:123]: this is no longer needed, you explicitly check for the 'config' field on line 69-73 now

File: pkg/project/v2/project_loader_test.go [hunk 3]
```diff
@@ -569,3 +570,21 @@ func getFullProjectLoaderContext(apis []string, projects []string, environments
 		ParametersSerde: config.DefaultParameterParsers,
 	}
 }
+
+func TestLoadProject(t *testing.T) {
+	t.Run("loading mixed projects", func(t *testing.T) {
```
Comment (UnseenWizzard) [inline pkg/project/v2/project_loader_test.go:575]: nit: if this test fails a month or two from now, I'll have no clue how "loading mixed projects" should work.
IMO would be nicer to name this more directly like the test projects "loading project with configs and account resources works" and "mixing configs and account resources in same file fails" - so the name alone tells me how things should have worked

**Review by UnseenWizzard — COMMENTED** [review]
looks ok overall, if the change to the error messages produced by the config loader doesn't mean we lose the information in which file things are wrong

File: pkg/persistence/config/loader/config_loader.go [hunk 4]
```diff
@@ -53,7 +56,32 @@ type singleConfigEntryLoadContext struct {
...
+			return nil, []error{newLoadError(filePath, ErrMixingConfigs)}
+		}
+
+		log.WithFields(field.F("file", filePath)).Warn("File %q appears to be an account resource file, skipping loading", filePath)
+		return []config.Config{}, nil
+	}
+
+	definedConfigEntries, err := parseFile(data)
 	if err != nil {
 		return nil, []error{newLoadError(filePath, err)}
```
[the remark this answers is not in the document]
Comment (Laubi) [reply pkg/persistence/config/loader/config_loader.go:86]: Yes

**Review by UnseenWizzard — DISMISSED** [review]
Thanks for addressing all comments 🎉 
Rebased to main and will merge as discussed.

**Review by UnseenWizzard — APPROVED** [review]

**Pull request merged**